### PR TITLE
Remote SSH [1/8]: Host registry + add/test host UI

### LINKED ISSUE #9: Remote SSH [1/8]: Host registry + add/test host UI
## Parent PRD

#8

## What to build

Introduce the foundational host-management layer: a `HostConnection` module that wraps SSH invocations and exposes a connection state machine, a `HostRegistry` persisted in `config.json`, and a host-management UI (add / edit / delete with a "Test connection" button).

This slice does not yet launch sessions on remote hosts or register remote projects — it only lets a user record that a host exists and verify cc-pewpew can reach it. See the Architecture, Data Model, and SSH Connection Shape sections of PRD #8 for the design.

## Acceptance criteria

- [ ] `hosts` table added to `config.json` persisting `{ hostId, alias, label }` entries
- [ ] `HostRegistry` module with CRUD over the hosts table
- [ ] `HostConnection` module encapsulating `ssh` invocation, with a state enum and an ssh-exit-signature parser (`auth-failed | network | dep-missing | unknown`)
- [ ] All path arguments passed through `HostConnection.exec` are shell-quoted safely (single-quote wrap + `'"'"'` escape)
- [ ] "Add host" / "Edit host" UI accepting `alias` and `label`
- [ ] "Test connection" button running `ssh -o BatchMode=yes -o ConnectTimeout=5 {alias} true`, reporting success or a specific error
- [ ] "Delete host" for the no-sessions case (cascade semantics land in a later slice)
- [ ] Unit tests for the ssh-exit-signature parser (table-driven, prior art: `src/main/diff-parser.test.ts`)
- [ ] No credentials of any kind stored in cc-pewpew
- [ ] Local project behavior unchanged

## Blocked by

None - can start immediately.

## User stories addressed

- User story 1
- User story 2
- User story 3
- User story 4
- User story 5
- User story 18
- User story 37
- User story 53

### PR DESCRIPTION
**Pull request by pmatos**
## Summary

- Adds a foundational host-management layer for the remote-SSH feature (slice 1/8 of PRD #8): a `HostRegistry` persisted in `config.json`, a `HostConnection` module that wraps `ssh` with a shell-quoting helper and a pure exit-signature parser, and a sidebar-accessible Manage Hosts dialog with Add / Edit / Delete / Test Connection.
- Local behaviour is unchanged — no session or project code is wired to `hostId` yet. Slice 2 will add `Session.hostId` and activate the delete-guard that is already in place.
- No credentials of any kind are stored; the registry schema is strictly `{ hostId, alias, label }`.

Closes #9.

## What's in it

- `src/shared/types.ts` — `HostId`, `Host`, `SshExitReason`, `TestConnectionResult`.
- `src/main/config.ts` — `hosts: Host[]` field in `AppConfig` (backfills to `[]` for existing configs via the existing spread-merge).
- `src/main/shell-quote.ts` + `shell-quote.test.ts` — POSIX single-quote wrapper (`single-quote wrap + '"'"'` escape) with 10 cases covering quoting edge cases.
- `src/main/ssh-exit-parser.ts` + `ssh-exit-parser.test.ts` — pure `classifySshExit({ exitCode, stderr })` returning `auth-failed | network | dep-missing | unknown`. Table-driven tests via `it.each` (20 cases) covering realistic OpenSSH stderr strings, multi-line `debug1:` prefixes, and the `exitCode: null` defensive path.
- `src/main/host-connection.ts` — `testConnection(alias)` using `execFile('ssh', ['-o','BatchMode=yes','-o','ConnectTimeout=5','--', alias, 'true'])` with a 15 s Node-side kill timeout, plus a forward-looking `exec(alias, argv)` for slice 2+ that applies `shellQuote` to every argv entry. Both paths pass `--` before the alias so a hand-edited config alias beginning with `-` cannot be interpreted by ssh as an option.
- `src/main/host-registry.ts` + `host-registry.test.ts` — CRUD with `validateAlias`/`validateLabel` (trim, non-empty, length bounds, reject leading `-`, whitespace, NUL), case-sensitive alias uniqueness (matches OpenSSH semantics), and a `deleteHost` guard that reads `sessions.json` and refuses when any persisted session has a matching `hostId`. 22 tests.
- `src/main/index.ts` — five new `ipcMain.handle` channels (`hosts:list`, `hosts:add`, `hosts:update`, `hosts:delete`, `hosts:test-connection`).
- `src/preload/index.ts` + `src/renderer/env.d.ts` — matching preload API surface (`listHosts`, `addHost`, `updateHost`, `deleteHost`, `testHostConnection`).
- `src/renderer/stores/hosts.ts` — Zustand store for the dialog: fetch, open/close, add/edit/delete, per-host test state + result, error surface.
- `src/renderer/components/ManageHostsDialog.tsx` — modal modeled on `BroadcastDialog`: row list with Test/Edit/Delete per host, add/edit form with Enter-to-submit / Escape-to-cancel, inline test-result pill (OK / "Network error: …" / "Auth failed: …"), inline error banner.
- `src/renderer/App.tsx` — adds a "🌐 Hosts (N)" button in the sidebar footer (as a sibling of the existing create-project ternary so it always renders), mounts `<ManageHostsDialog />` before `<StatusBar />`, and fetches hosts on startup.
- `src/renderer/styles/global.css` — `.hosts-dialog*`, `.hosts-row`, `.host-label`, `.host-alias`, `.host-test-result.ok/err`, `.hosts-error`, `.hosts-form` styles (copy-renamed from the existing `.broadcast-dialog` rules; no new design tokens).

## Security notes

- `validateAlias` rejects leading `-`, embedded whitespace, and NUL; `testConnection` and `exec` additionally pass `--` before the alias to defend against bypass (e.g. hand-edited `config.json`).
- The `HostConnection` module comments the invariant that callers of `exec` pass path arguments as individual argv entries (each wrapped by `shellQuote`), never as pre-assembled shell strings.
- `BatchMode=yes` prevents any interactive ssh prompts during "Test connection"; a 15 s Node-side kill timeout backs up ssh's own `ConnectTimeout=5` against DNS hangs.

## Test plan

- [x] `npx tsc --noEmit` — clean
- [x] `npx vitest run` — 103/103 pass (new: 10 shell-quote + 20 ssh-exit-parser + 22 host-registry)
- [x] `npx eslint .` — no new violations (one pre-existing `no-empty` at `src/main/index.ts:201` is unrelated)
- [x] `npx electron-vite build` — production build succeeds
- [x] CDP smoke: Hosts button opens dialog; add with alias `persist-check` / label `Persist` creates a row and the sidebar counter updates to `🌐 Hosts (1)`; Test connection on a non-existent alias runs real `ssh` and surfaces `"Network error: ssh: Could not resolve hostname …"`; attempting to re-add the same alias surfaces `"Alias already exists"` and the row count stays at 1; Delete removes the row and the counter returns to `🌐 Hosts (0)`; `~/.config/cc-pewpew/config.json` shows the expected `{ hostId, alias, label }` record during the test and empty array after cleanup.

## Out of scope (per PRD #8)

- Sessions / projects do not yet carry `hostId` — slice 2.
- No persistent ControlMaster / reverse-socket tunnel — slice 3–4.
- No remote project registration, bootstrap script install, thumbnails, or review pane — later slices.

### CODE REVIEW

File: src/main/host-connection.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+function runSsh(argv: string[], timeoutMs: number): Promise<SshRunResult> {
+  return new Promise((resolve) => {
+    const child = execFile(
+      'ssh',
+      argv,
+      { timeout: timeoutMs, maxBuffer: 64 * 1024 },
+      (error, stdout, stderr) => {
+        const timedOut = Boolean(error && (error as NodeJS.ErrnoException).code === 'ETIMEDOUT')
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/main/host-connection.ts]: Fixed in `48255ba`. `runSsh` now treats `error.killed === true` as a timeout (the check that actually fires when `execFile`'s `timeout` option kills the child with SIGTERM); the `ETIMEDOUT` check is kept for belt-and-braces. Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).

File: src/renderer/components/ManageHostsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+      setSubmitting(false)
+    }
+  }
+
+  const handleKey = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      e.preventDefault()
+      void handleSubmit()
+    } else if (e.key === 'Escape') {
+      onCancel()
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/renderer/components/ManageHostsDialog.tsx:56]: Fixed in `48255ba`. The document-level Escape listener now bails out when an Add/Edit form is open (reads `useHostsStore.getState()` at keydown time) so only the form's own handler fires. Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).

File: src/renderer/components/ManageHostsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    const handler = (e: KeyboardEvent) => {
+      if (e.key !== 'Escape') return
+      // When an Add/Edit form is open, let the form's own Escape handler cancel
+      // the form. React synthetic events don't stop native propagation, so
+      // without this guard both handlers fire and the whole dialog closes.
+      const state = useHostsStore.getState()
+      if (state.addingNew || state.editingHostId !== null) return
+      closeDialog()
+    }
+    document.addEventListener('keydown', handler)
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/renderer/components/ManageHostsDialog.tsx:164]: Fixed in `220eaf9`. The dialog's Escape listener now registers in the capture phase and calls `e.stopPropagation()`, so it runs before `App.tsx`'s bubble-phase global handler and prevents it from also firing. No more side-effect clearing of `activeSessionId` / session selection when dismissing the modal. Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).

File: src/renderer/components/ManageHostsDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    // as a side effect of dismissing the modal).
+    const handler = (e: KeyboardEvent) => {
+      if (e.key !== 'Escape') return
+      const state = useHostsStore.getState()
+      // When an Add/Edit form is open, let the form's own Escape handler cancel
+      // the form. Still stop propagation so App's global handler doesn't also
+      // fire. React synthetic events don't stop native propagation, so the
+      // form's handler alone isn't enough.
+      e.stopPropagation()
+      if (state.addingNew || state.editingHostId !== null) return
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/renderer/components/ManageHostsDialog.tsx]: Fixed in `6ade00c`. Capture phase + `stopPropagation` was structurally wrong — either it killed the form's own Escape (as you noted) or it leaked through to App.tsx's side effects. Reverted to a bubble-phase listener and made `App.tsx`'s global Escape early-return when `useHostsStore.getState().dialogOpen` is true. The modal now fully owns Escape: form-Escape cancels the form, empty-form Escape closes the dialog, App.tsx doesn't fire in either case. Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).

File: src/main/host-connection.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        // execFile's `timeout` option kills the child with `killSignal` (default
+        // SIGTERM) when it fires. Node sets `error.killed === true` in that
+        // case; `error.code === 'ETIMEDOUT'` only appears for OS-level socket
+        // timeouts and is kept as a belt-and-braces check.
+        const errno = (error ?? null) as (NodeJS.ErrnoException & { killed?: boolean }) | null
+        const timedOut = Boolean(errno && (errno.killed === true || errno.code === 'ETIMEDOUT'))
+        const code = error
+          ? typeof (error as { code?: unknown }).code === 'number'
+            ? (error as { code: number }).code
+            : (child.exitCode ?? 1)
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/main/host-connection.ts:51]: Fixed in `6ade00c`. `runSsh` now detects `error.code === 'ENOENT'` specifically and synthesizes an exit-127 result with stderr `"ssh: command not found"`, which `classifySshExit` already routes to `dep-missing`. Users without OpenSSH will now see "Missing dep" in the test-connection pill instead of a generic "Failed". Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).

File: src/renderer/components/ManageHostsDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+      setSubmitting(false)
+    }
+  }
+
+  const handleKey = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      e.preventDefault()
+      void handleSubmit()
+    } else if (e.key === 'Escape') {
+      onCancel()
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/renderer/components/ManageHostsDialog.tsx:56]: Fixed in `b9cb159`. You correctly identified the race: Zustand's synchronous `set` in `cancelEdit` cleared `addingNew`/`editingHostId` before the document-level handler could check them. Replaced the store-state guard with a DOM-target guard — `target.closest('.hosts-form')` — so the document handler skips when the Escape originated inside an Add/Edit form, independent of any state race. Confirmed by two independent AI reviewers (Claude Opus 4.6 and GPT-5.4).
